src/legacy/components/TransactionConfirmationModal/LegacyConfirmationPendingContent.tsx

```diff
@@ -0,0 +1,251 @@
+import checkImage from 'legacy/assets/cow-swap/check.svg'
+import { MediumAndUp, useMediaQuery } from 'legacy/hooks/useMediaQuery'
+import { ExternalLink } from 'legacy/theme'
+import { shortenAddress } from 'legacy/utils'
+import { getChainCurrencySymbols } from 'legacy/utils/gnosis_chain/hack'
+
+import { getStatusIcon } from 'modules/account/containers/AccountDetails'
+import { useGnosisSafeInfo, useWalletDetails, useWalletInfo } from 'modules/wallet'
+import { getIsMetaMask } from 'modules/wallet/api/utils/connection'
+import { getWeb3ReactConnection } from 'modules/wallet/web3-react/connection'
+import { walletConnectConnection } from 'modules/wallet/web3-react/connection/walletConnect'
+
+import {
+  ApproveComparison,
+  ApproveFooter,
+  ApproveWrapper,
+  CloseIconWrapper,
+  CompareItem,
+  ItemList,
+  LowerSection,
+  StepsIconWrapper,
+  StepsWrapper,
+  UpperSection,
+  WalletIcon,
+  Wrapper,
+} from './styled'
+import { ConfirmOperationType } from './types'
+
+enum WalletType {
+  SAFE,
+  SC,
+  EOA,
+}
+
+export function getOperationMessage(operationType: ConfirmOperationType, chainId: number): string {
+  const { native, wrapped } = getChainCurrencySymbols(chainId)
+
+  switch (operationType) {
+    case ConfirmOperationType.WRAP_ETHER:
+      return 'Wrapping ' + native
+    case ConfirmOperationType.UNWRAP_WETH:
+      return 'Unwrapping ' + wrapped
+    case ConfirmOperationType.APPROVE_TOKEN:
+      return 'Approving token'
+    case ConfirmOperationType.ORDER_CANCEL:
+      return 'Canceling your order'
+    case ConfirmOperationType.REVOKE_APPROVE_TOKEN:
+      return 'Revoking token approval'
+    case ConfirmOperationType.CONVERT_VCOW:
+      return 'Converting vCOW to COW'
+    case ConfirmOperationType.CLAIM_VESTED_COW:
+      return 'Claiming vested COW'
+    default:
+      return 'Almost there!'
+  }
+}
+
+function getOperationLabel(operationType: ConfirmOperationType): string {
+  switch (operationType) {
+    case ConfirmOperationType.WRAP_ETHER:
+      return t`wrapping`
+    case ConfirmOperationType.UNWRAP_WETH:
+      return t`unwrapping`
+    case ConfirmOperationType.APPROVE_TOKEN:
+      return t`token approval`
+    case ConfirmOperationType.REVOKE_APPROVE_TOKEN:
+      return t`revoking token approval`
+    case ConfirmOperationType.ORDER_SIGN:
+      return t`order`
+    case ConfirmOperationType.ORDER_CANCEL:
+      return t`cancellation`
+    case ConfirmOperationType.CONVERT_VCOW:
+      return t`vCOW conversion`
+    case ConfirmOperationType.CLAIM_VESTED_COW:
+      return t`vested COW claim`
+  }
+}
+
+function getSubmittedMessage(operationLabel: string, operationType: ConfirmOperationType): string {
+  switch (operationType) {
+    case ConfirmOperationType.ORDER_SIGN:
+      return t`The order is submitted and ready to be settled.`
+    default:
+      return `The ${operationLabel} is submitted.`
+  }
+}
+
+function getWalletNameLabel(walletType: WalletType): string {
```

Comment:
Should this be here or in wallets module?

Reply:
Same as here https://github.com/cowprotocol/cowswap/pull/2571#discussion_r1211653921